eclipse-theia/theia
# [ci] run dash-licenses as part of GH workflow

#### What it does

Add a "3PP FOSS license check" GitHub workflow, that uses
the Eclipse Foundation dash-licences tool to validate the
licenses of our npm dependencies.

One can also run the license check locally with command:
"yarn license:check"

I have considered 3 alternatives:

(1) add the license check as a step of the build job of the build workflow. 

The license check step needs to report
success or it will fail the build job and the publish job
(when it applies). If the license check reports success, it's
burried in the multiple steps of its job, not discoverable
from the PR page.

(2) add the license check as a new job in the build workflow.

This works better but there is redundency in the job -  we need
to re-install node, checkout the repo, etc.

One advantage of this approach is that the license check has
visibility in the PR's CI checks section and can be made to fail
without impacting the other jobs.

(3) (chosen) add a new workflow for the license check

Like option 2, this gives the license check visibility
in the PR's CI checks section and the job can fail without impacting
other workflows/jobs.

This option has the additional advantage that it's quite
generic (for Theia-based repos) and could be re-used more
easily, without changing the main workflow of the adopting repo.


#### How to test

- Run the license check locally: 
`yarn license:check` and verify that ~38 dependencies are found that need attention
- Look at this PR's CI steps and verify that `3PP License Check` fails, and in the log, lists the same dependencies


#### Review checklist


#### Reminder for reviewers

- as a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)
